pdx213: amend DT by a couple more nodes

erebion opened the pull request:
A couple more nodes.

Not sure what I'll need to improve.

Commit:
arm64: dts: qcom: sm6350-lena: Enable ADSP, CDSP & MPSS

Enable the ADSP, CDSP and MPSS (Modem) nodes for  for the Xperia 10 III.

Signed-off-by: erebion <erebion@erebion.eu>

Commit:
arm64: dts: qcom: sm6350-lena: Enable IPA

Enable the IPA node for the Xperia 10 III which makes mobile data work.

Signed-off-by: erebion <erebion@erebion.eu>

Commit:
arm64: dts: qcom: sm6350-lena: Enable wifi

Configure regulators and enable it for the Xperia 10 III

Signed-off-by: erebion <erebion@erebion.eu>

Commit:
arm64: dts: qcom: sm6350-lena: Enable Bluetooth

Configure the required regulators for Bluetooth and enable it on the
Xperia 10 III.

Signed-off-by: erebion <erebion@erebion.eu>

Commit:
arm64: dts: qcom: sm6350-lena: Enable PM7250B

Enable PM7250B and some components on it for the Xperia 10 III.

Signed-off-by: erebion <erebion@erebion.eu>

Commit:
arm64: dts: qcom: sm6350-lena: Enable PMK8350

Configure and enable nodes for PMK8350 on the Xperia 10 III.

Signed-off-by: erebion <erebion@erebion.eu>

erebion commented:
Huh, comparing with sm7325-nothing-spacewar.dts, which looks different for the fsa4480 USB-C stuff, perhaps this isn't even correct...

erebion commented:
> Huh, comparing with sm7325-nothing-spacewar.dts, which looks different for the fsa4480 USB-C stuff, perhaps this isn't even correct...

Yeah... This is incorrect. I have no idea how to put in the fs4480 stuff correctly.

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -26,6 +26,7 @@
 	qcom,board-id = <0x1000B 0>;
 
 	aliases {
+		serial0 = &uart9;
```
uart9 is disabled in sm6350.dtsi so if you wanna use it, enable it.

Is debug UART actually exposed somewhere on the board? Did you test it?

erebion replied:
I haven't been able to test it, as the contacts are in the SIM slot and I have no source for an adapter: https://developer.sony.com/open-source/aosp-on-xperia-open-devices/guides/access-uart-ports

erebion replied:
Is it just adding this?

```
&uart9 {
        status = "okay";
};
```

erebion replied:
Correction: It's in the combined microSD/SIM slot, on two microSD contacts, so I'd need a fake microSD card with two wires coming out the phone.

z3ntu replied:
If you can't test it, leave it disabled.

erebion replied:
I will test it, I'm currently looking for a Pinephone microSD extender, which would let me connect to the pins and and connect the wires to the extender:

https://pine64.com/product/pinephone-microsd-extender/?v=0446c16e2e66

Just need to find out where to get one...

erebion replied:
I've ordered an adapter and will probably be able to test this week. Let me just quickly test and I'll comment again after that's done. :)

erebion replied:
I've dropped UART for now. I will add it again later should I find out how it gets turned on. I'd just use `strings` on the images to look for fastboot commands, but Sony does not provide the images, only their flashing tool, annoyingly.

z3ntu replied:
sounds good

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -4,12 +4,19 @@
 
+/* PM7250B is configured to use SID2/3 */
+#define PM7250B_SID 2
+#define PM7250B_SID1 3
```
Please split everything in the "Add Battery" into a separate commit that can go upstream already. So everything apart from battery, charger, fuel qgauge essentially.

erebion replied:
Which nodes/things should I group into how many separate commits..?

z3ntu replied:
split the pm7250b commit in two, one for general enablement, and one for the battery stuff

erebion replied:
I've split the commits, hopefully it is correct now. Took me a while to get around to it, health and all that.

Anyway, needed a couple tries to split the commits correctly. Learned something new.

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -383,6 +383,11 @@
 
+&mpss {
+	firmware-name = "qcom/sm6350/sony/pdx213/modem.mbn";
+	status = "okay";
```
newline before status

erebion replied:
done

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -475,6 +513,73 @@
+		vddio-supply = <&pm6350_l11>;
+		vddxo-supply = <&pm6350_l7>;
+		vddrf-supply = <&pm6150l_l2>;
+		vddch0-supply = <&pm6150l_l10>;
```
You should probably use the power sequencing (pwrseq) version of these bindings. IIRC the current ones are marked as deprecated?

erebion replied:
power sequencing..? link pls?

z3ntu replied:
`Documentation/devicetree/bindings/regulator/qcom,qca6390-pmu.yaml` but not sure it's easy to enable without extra changes. So maybe keep it as-is if it works, and maybe if during upstream review somebody says something you can take a look

erebion replied:
I'm not sure what part of the file you're referring to, the term `sequencing` does not show up, neither does `pwrseq`.

z3ntu replied:
Instead of putting the supplies into the bluetooth node, you make a new PMU node (see the bindings I referenced) and then use the vregs from there to use in the bluetooth node. See cca4fe34979a77d9df5c0acf935e9cc939fe5cd6

But as I said before, it's probably fine for now as-is if it works. We can change it later as well, or when somebody asks in upstream review.

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -330,6 +330,13 @@
+	qcom,gsi-loader = "self";
+	memory-region = <&pil_ipa_fw_mem>;
+	firmware-name = "qcom/sm6350/sony/pdx213/ipa_fws.mbn";
+	status = "okay";
```
newline before status. Also put qcom,gsi-loader below the other "generic properties" (memory-region & firmware-name).

erebion replied:
done

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -292,6 +292,11 @@
 
+&cdsp {
+	firmware-name = "qcom/sm6350/sony/pdx213/cdsp.mbn";
+	status = "okay";
```
newline before status

erebion replied:
done

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -64,6 +64,11 @@
 
+&adsp {
+	firmware-name = "qcom/sm6350/sony/pdx213/adsp.mbn";
+	status = "okay";
```
newline before status

erebion replied:
done

z3ntu commented:
I also have some notes on the commit messages, not sure I can (easily) comment on those with github UI, mailing lists and Gerrit make this much more easy :upside_down_face: 

But essentially, when you're enabling the nodes in your device dts, don't call it "add node" but something like "configure the required regulators for wifi and enable the node" because you're enabling it and not adding it (the node was already added to sm6350.dtsi and friends before)

erebion commented:
By the way, should all this be enough to plug in a USB stuck and have it recognised? If so, there might still be something missing... Perhaps a config option I hadn't set?

If not, then fine. :)

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -458,8 +768,79 @@
+	status = "okay";
+
+	bluetooth {
+		compatible = "qcom,wcn3991-bt"; // Downstream DT mentions a WCN3990, but this seems incorrect
```
according to my info SM7225 can be paired with either WCN3991 or WCN3988. Not sure about SM6350 but it's probably the same as well given it's essentially just two speedbins of the same SoC.

erebion replied:
Then it is WCN3991, as the version reply has the right length for WCN3991, but wouldn't work for WCN3900 which has a different length and the same as WCN3988.

I'll remove the comment later when I get to it.

Commit:
arm64: dts: qcom: sm6350-lena: Enable battery and Type-C handling

Configure USB-C plumbing and configure battery charging & fuel gauge for
the Xperia 10 III.

Signed-off-by: erebion <erebion@erebion.eu>

erebion commented:
> mailing lists and Gerrit make this much more easy

Everytime I try to follow the mailing list, I give up after a while. No idea how to follow that all along and keep track of the right things.

For example, let's say I want to know what sdm670 stuff gets upstreamed (which I actually would like to know), how would I even find that and where? It's all very confusing.

erebion commented:
@z3ntu Anything else? Took me a while to get back to it, but I think I've addressed all points now. :)

z3ntu commented:
I've squashed ADSP, CDSP and modem together, cleaned up some `./scripts/checkpatch.pl` warnings, and some commit messages a bit. You can check what it looks like now maybe.

z3ntu commented:
I'll try to merge this asap, don't spot any obvious problem that should be fixed

z3ntu reviewed (approved):
LGTM! Looking forward to future contributions, and don't forget to send your patches upstream sometime soon!

erebion commented:
> LGTM! Looking forward to future contributions

Also looking forward to future contributions! Couldn't do this if I didn't have someone point out what I need to change. Thanks! :)

>and don't forget to send your patches upstream sometime soon!

What do I have to do? And how? I've heard people get pretty unfriendly if there are tiny punctuation typos in the patch emails, so I'll probably need to be careful.

Anyway, will be able to get to that when health improves in a couple of days.

z3ntu commented:
> What do I have to do? And how? I've heard people get pretty unfriendly if there are tiny punctuation typos in the patch emails, so I'll probably need to be careful.

Don't worry about it! If you repeatedly ignore review feedback people give you, they may get a bit annoyed at you but otherwise it's no problem :) Just like any human interaction really.

I can also guide you through it a bit - meaning you can send the patch series to me via email first to make sure it looks correct - you'll definitely want to use this tool to manage your patch series which makes it way easier https://b4.docs.kernel.org/en/latest/ , check https://b4.docs.kernel.org/en/latest/contributor/overview.html The docs for that tool are fairly good I think.

erebion commented:
>The docs for that tool are fairly good I think.

It lists a lot of config options and parameters and things with unfamiliar names such as 

>For example, to set the `b4.midmask` option, add the following section to the relevant git config file:

But no idea what I want to do with it.

I've found `b4 prep`, which lools like something I might want to try, but it just exits without doing anything and it does not seem to do anything.

Idk... Why has sending in a patch have to be harder than enabling the flashlight LED?

Cannot figure out what to do with the tool.

erebion commented:
>Cannot figure out what to do with the tool.

What are some of the commands you use when upstreaming your changes?

Maybe it's easier to figure this out by seeing an example of how it is used rather than trying to figure out what parameters could be useful for whatever I should be doing.

z3ntu commented:
These are the rough steps: https://b4.docs.kernel.org/en/latest/contributor/overview.html#what-s-the-b4-contributor-workflow

You first make a branch using `b4 prep -n YOUR_TOPIC_NAME` (e.g. `b4 prep -n pdx213-bringup`), then depending on where you start, you git rebase and put the cover letter *before* the first change you wanna send - or you cherry-pick changes on top. You can always see what happens by running e.g. `b4 send -o /tmp/presend` and viewing the mail files it has written to that directory (they don't get cleaned by a subsequent run, but overwritten if the ordering+subject is the same, so run `rm /tmp/presend/*` sometimes)

And the rest of that docs page should have the most important other steps. For `b4 send` there's some options you can use before sending it public, e.g. the `--reflect` or `--preview-to` (if I don't have them written wrong)
